core-foundation/src/array.rs

```diff
@@ -218,14 +216,14 @@ mod tests {
             assert_eq!(iter.len(), 5);
 
             for elem in iter {
-                let number: CFNumber = TCFType::wrap_under_get_rule(mem::transmute(elem));
+                let number: CFNumber = TCFType::wrap_under_get_rule(mem::transmute(elem.as_concrete_TypeRef()));
                 sum += number.to_i64().unwrap()
             }
 
             assert!(sum == 15);
 
             for elem in arr.iter() {
-                let number: CFNumber = TCFType::wrap_under_get_rule(mem::transmute(elem));
+                let number: CFNumber = TCFType::wrap_under_get_rule(mem::transmute(elem.as_concrete_TypeRef()));
```

Comment:
Why are these changes necessary?

Reply:
Ehm.. They are not, any longer. they were before when I actually changed the structs to contain a bool, then the transmute did not work because the types were of different size. I'll remove it! It's removed in the commit after anyway, but anyhow.